untyped-lambda-calculus/article.md

```diff
@@ -146,63 +146,73 @@ This brings us to the discussion of evaluation strategies.
 
-It's interesting to note that the call-by-value strategy corresponds to eager evaluation.
+The weak strategy corresponding to the normal order is called _call-by-name_, and the one corresponding to the applicative order is _call-by-value_.
+Call-by-value corresponds to eager evaluation.
 Call-by-name, on the other hand, corresponds to lazy evaluation.
+
 In eager evaluation, arguments are always evaluated before the function call.
 Most programming languages follow this strategy by default.
 In lazy evaluation, some arguments are potentially never evaluated at all, but some computations may be duplicated.
 Very few programming languages implement lazy evaluation this way (the only real example that comes to mind is Algol 60).
 
-A practically significant variation on call-by-name is _call-by-need_, which is, at least in theory, used in Haskell.
+A more practical variation of call-by-name is _call-by-need_, which is, at least in theory, used in Haskell.
 Call-by-need behaves exactly like call-by-name, except that reduction steps that would duplicate computations, don't.
-The trick is to share the computation corresponding to an argument everywhere it appears, such that the shared computation is performed at most once.
+It does that by sharing the computation corresponding to an argument everywhere it appears, so any shared computation is performed at most once.
 
-Expressions that are not reducible using the chosen evaluation strategy are said to be in the _normal form_.
+To sum up, here's a table with all the mentioned strategies and their characteristics:
 
-If a given term has a normal form under both strong strategies, it's the same for both.
-Note that applicative order strategy is strictly weaker than normal order strategy.
-Some terms that normalize under normal order don't under applicative order.
-Likewise for the weak strategies.
+| Name                 | Outermost- or innermost- first? | Strong or weak? |
+|----------------------|---------------------------------|-----------------|
+| Normal order         | Outermost                       | Strong          |
+| Applicative order    | Innermost                       | Strong          |
+| Call-by-name (eager) | Outermost                       | Weak            |
+| Call-by-value (lazy) | Innermost                       | Weak            |
+| Call-by-name         | Innermost                       | Weak            |
 
-Furthermore, if a given term has a normal form at all, then reduction under the normal order strategy will always reach it.
-We will discuss terms that do not have a normal form a little later.
-
-Speaking of reductions, let's also mention _eta_-reduction ($\eta$-reduction).
-The rule of eta-reduction says that $\lambda x. f\;x = f,$ if $x$ does not appear free in $f$.
-It boils down to the statement that two functions are the same if and only if they give the same result for all possible arguments.
```

Comment:
Okay, I didn't find a good place to mention this (besides here, I mean), but it is important to mention this, because the term η-reduction often gets thrown around when discussing Haskell programs.